network/ipfs_impl_test.go

```diff
@@ -579,12 +571,75 @@ func testNetworkCounters(t *testing.T, n1 int, n2 int) {
 	ctxto, cancelto := context.WithTimeout(ctx, 5*time.Second)
 	defer cancelto()
 	ctxwait, cancelwait := context.WithCancel(ctx)
-	defer cancelwait()
 	go func() {
-		wg1.Wait()
-		wg2.Wait()
+		// Wait until all streams are closed
+		throttler := time.NewTicker(time.Millisecond * 5)
+		defer throttler.Stop()
+		for {
+			h1.lk.Lock()
+			var done bool
+			for _, s := range h1.streams {
+				s.lk.Lock()
+				closed := s.closed
+				closed = closed || s.err != nil
+				s.lk.Unlock()
+				if closed {
+					continue
+				}
+				pid := s.Protocol()
+				for _, v := range bsnet.DefaultsProtocols {
+					if pid == v {
+						goto ElseH1
+					}
+				}
+			}
+			done = true
+		ElseH1:
+			h1.lk.Unlock()
+			if done {
+				break
+			}
+			select {
+			case <-ctxto.Done():
+				return
+			case <-throttler.C:
+			}
+		}
```

Comment:
this makes me sad :(   (but not worth doing it any other way that I can see)

Reply:
Why can't we use the `EventBus` interface with `event.EvtPeerConnectednessChanged{}` here?

Reply:
Because its not synchronous